doc/api/errors.md

```diff
@@ -2706,6 +2706,183 @@ size.
 This `Error` is thrown when a read is attempted on a TTY `WriteStream`,
 such as `process.stdout.on('data')`.
 
+<a id="openssl-error-codes"></a>
+## OpenSSL Error Codes
+
+<a id="Time Validity Errors"></a>
+### Time Validity Errors
+
+<a id="CERT_NOT_YET_VALID"></a>
+#### `CERT_NOT_YET_VALID`
+
+The certificate is not yet valid: the notBefore date is after the current time.
+
+<a id="CERT_HAS_EXPIRED"></a>
+#### `CERT_HAS_EXPIRED`
+
+The certificate has expired: that is the notAfter date is before the current
+time.
+
+<a id="CRL_NOT_YET_VALID"></a>
+#### `CRL_NOT_YET_VALID`
+
+The revocation data have a future issue date.
+
+<a id="CRL_HAS_EXPIRED"></a>
+#### `CRL_HAS_EXPIRED`
+
+The certificate revocation list (CRL) has expired.
+
+<a id="CERT_REVOKED"></a>
+#### `CERT_REVOKED`
+
+The certificate has been revoked; it is on a certificate revocation list (CRL).
+
+<a id="Trust or Chain Related Errors"></a>
+### Trust or Chain Related Errors
+
+<a id="UNABLE_TO_GET_ISSUER_CERT"></a>
+#### `UNABLE_TO_GET_ISSUER_CERT`
+
+The issuer certificate of a looked up certificate could not be found. This
+normally means the list of trusted certificates is not complete.
+
+<a id="UNABLE_TO_GET_ISSUER_CERT_LOCALLY"></a>
+#### `UNABLE_TO_GET_ISSUER_CERT_LOCALLY`
+
+The certificate’s issuer is not known. This is the case if the issuer is not
+included in the trusted certificate list.
+
+<a id="DEPTH_ZERO_SELF_SIGNED_CERT"></a>
+#### `DEPTH_ZERO_SELF_SIGNED_CERT`
+
+The passed certificate is self-signed and the same certificate cannot be found
+in the list of trusted certificates.
+
+<a id="SELF_SIGNED_CERT_IN_CHAIN"></a>
+#### `SELF_SIGNED_CERT_IN_CHAIN`
+
+The certificate’s issuer is not known. This is the case if the issuer is not
+included in the trusted certificate list.
+
+<a id="CERT_CHAIN_TOO_LONG"></a>
+#### `CERT_CHAIN_TOO_LONG`
+
+The certificate chain length is greater than the supplied maximum depth.
```

Comment:
```suggestion
The certificate chain length is greater than the maximum depth.
```
"supplied" suggests it's configurable but it's not (in node - no binding for `X509_STORE_CTX_set_depth()`.)